src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.Helpers.cs

```diff
@@ -9,19 +9,22 @@ namespace System.Text.Json
 {
     public static partial class JsonSerializer
     {
-        private static void WriteCore<TValue>(Utf8JsonWriter writer, TValue value, Type inputType, JsonSerializerOptions options)
+        private static void WriteCore<TValue>(
+            Utf8JsonWriter writer,
+            in TValue value,
+            Type inputType,
+            JsonSerializerOptions options)
         {
             Debug.Assert(writer != null);
 
             //  We treat typeof(object) special and allow polymorphic behavior.
-            if (inputType == typeof(object) && value != null)
+            if (value != null && inputType == typeof(object))
```

Comment:
Is it worth it to cache `(typeof(object))`? https://sharplab.io/#v2:C4LghgzgtgNAJiA1AHwAICYCMBYAUKgZgAIMiBhIgbzyNqIAcAnASwDcxgBTIgFQE963APoB7AEYArTgGNg/QUQC8RYAM4iAZgApxU2QEoA3DTonahImJEiANkQCyW+d1WD9SgHwq1S5a/XaujLARmZEYRZWtkQAck4+/u6KXv6+RKKSwc7GuAC+QA==

Reply:
Thanks. That is somewhat unfortunate, but yes it makes sense to cache it then. Adding to JsonClassInfo.